pkg/wallet/internal/mapping/mapping_send_with_results.go

```diff
@@ -0,0 +1,49 @@
+package mapping
+
+import (
+	"fmt"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/bsv-blockchain/go-sdk/chainhash"
+	sdk "github.com/bsv-blockchain/go-sdk/wallet"
+	"github.com/go-softwarelab/common/pkg/slices"
+	"github.com/go-softwarelab/common/pkg/to"
+)
+
+func MapSendWithResultsFromWDKToSDK(sendWithResults []wdk.SendWithResult) ([]sdk.SendWithResult, error) {
+	result, err := slices.MapOrError(sendWithResults, mapSendWithResultWDKtoSDK)
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare SendWithResult list: %w", err)
+	}
+	return result, nil
+}
+
+func mapSendWithResultWDKtoSDK(sendWithResult wdk.SendWithResult) (sdk.SendWithResult, error) {
+	txID, err := chainhash.NewHashFromHex(sendWithResult.TxID.String())
+	if err != nil {
+		return sdk.SendWithResult{}, fmt.Errorf("cannot restore tx id from hex %s, %w", sendWithResult.TxID, err)
+	}
+
+	status, err := mapSendWithResultStatusWDKtoSDK(sendWithResult.Status)
```

Comment:
Hmmm... just thinking

```suggestion
	status, err := SendWithResultStatusMapper.Map(status)
```

and:

```golang
type EnumMapper[From, To comparable] struct {
	lookup map[From]To
	name string
}

func NewStatusMapper[From, To comparable](name string, mappings map[From]To) *EnumMapper[From, To] {
	return &EnumMapper[From, To]{
		lookup: mappings,
		name:   name,
	}
}

func (mapper *EnumMapper[From, To]) Map(status From) (To, error) {
	if mappedStatus, ok := mapper.lookup[status]; ok {
		return mappedStatus, nil
	}
	return to.ZeroValue[To](), fmt.Errorf("unexpected %s: [%T] %v", mapper.name, status, status)
}

var SendWithResultStatusMapper = NewStatusMapper(
	"sendWithResult status from storage",
	map[wdk.SendWithResultStatus]sdk.ActionResultStatus{
		wdk.SendWithResultStatusUnproven: sdk.ActionResultStatusUnproven,
		wdk.SendWithResultStatusSending:   sdk.ActionResultStatusSending,
		wdk.SendWithResultStatusFailed:    sdk.ActionResultStatusFailed,
	},
)

```

Reply:
1. Doesn't look like a big benefit - from "effective" 5 lines of code - now there is 20 and the reduction in future doesn't seem to be greater
2. Using short map instead of switch has impact on the speed of this mapping - I'm learning pre-optimisation from you 😂

Reply:
The reason I'm suggesting this is because I've already seen similar mapping functions. 
I'm also guessing, that the number of such mappings will grow - so handy function might be beneficial.
But it's minor anyway, and doesn't directly concerns your PR so - do whatever you decide :) 

https://github.com/bsv-blockchain/go-wallet-toolbox/blob/4841f9e24ba5ec46bf8044d32f0272203a18af51/pkg/wallet/internal/mapping/mapping_send_with_results.go#L38

https://github.com/bsv-blockchain/go-wallet-toolbox/blob/62488744b7486d522fd98d9514aeac0595f61db3/pkg/wallet/internal/mapping/mapping_list_actions.go#L125 
- In the second example, it also should be wdk type (instead of string)

Reply:
Sorry, I don't want to change it, I don't really see a big benefit from making it that way.